src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -625,7 +613,6 @@ private void ProcessError(Exception error)
             foreach (var actor in involvedStageActors)
             {
                 actor.Tell(new Status.Failure(error));
-                _requests = _requests.Remove(actor);
```

Comment:
Do we not need to track requests anymore?

Reply:
In the event of a serialization error, the KafkaConsumerActor does not get killed. Removing all requests will make the consumer actor and the downstream stage to hang if the user decides to ignore the serialization error.

Reply:
The `actor` in this case was the `StageActor` for the graph stage though, was it not?

Reply:
No, its the KafkaConsumerActor, the StageActor are only used as a middle man to pass messages back from the KafkaConsumerActor to the stage using its Receive method. StageActor will never be killed in an exception.

Reply:
Why would the `KafkaConsumerActor` track requests for itself?

Reply:
They are used to track the requests made from the stages, only one for a normal stage, multiple for subsource stages.
When a stage is pulling, it sends a RequestsMessages message to the consumer actor which contains the topic and partition the stage is interested in. The consumer actor uses these to route messages it received from kafka to the interested stages.

Reply:
That would mean that the `actor` in this case is the `StageActor` then right? Otherwise this code would make no sense.

Reply:
i.e. the `HashSet<T>` would max out at 1 item and there wouldn't be any purpose in supporting tracking requests from multiple stages.

Reply:
Oh, right, yes, `actor` in this context is the StageActor. `KafkaConsumerActor` is sending all of the messages to the stage `StageActor`, the stage registers a `Receive` on their `StageActor` to read all those messages

Reply:
Ok, got it confirmed in our chat - these are indeed requests being tracked to individual `StageActor`s, which makes sense.